src/libstd/sys/unix/ext/fs.rs

```diff
@@ -32,8 +32,27 @@ pub trait FileExt {
     ///
     /// The current file cursor is not affected by this function.
     ///
-    /// Note that similar to `File::read`, it is not an error to return with a
+    /// Note that similar to [`File::read`], it is not an error to return with a
     /// short read.
+    ///
+    /// [`File::read`]: ../../../../std/fs/struct.File.html#read.v
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::os::unix::prelude::FileExt;
+    /// use std::fs::File;
+    ///
+    /// # use std::io;
+    /// # fn f() -> io::Result<()> {
+    /// let buf = &mut [0u8; 8];
+    /// let file = File::open("foo.txt")?;
+    ///
+    /// // We now read 8 bytes from the offset 10.
```

Comment:
This will only read up to 8 bytes so I think it's important to do something with the value `read_at` returns in this example. The same goes for `write_at`.

Reply:
Good point.

Reply:
maybe something like this?

```rust
// Read up to 8 bytes from the offset 10.
let num_bytes_read = file.read_at(buf, 10)?;
println!("read {} bytes: {:?}", num_bytes_read, buf);
```